### PR TITLE
Add `/compatible-libraries` route

### PR DESCRIPTION
**Pull request by lhoestq**
- Renamed `dataset-loading-tags` -> `dataset-compatible-libraries` in code
- Added migration for queue and cache
- Added route

### CODE REVIEW

File: services/api/src/api/config.py [hunk 1]
```diff
@@ -88,6 +88,7 @@ class EndpointConfig:
                 "split": "split-is-valid",
             },
             "/statistics": {"split": "split-descriptive-statistics"},
+            "/compatible-libraries": {"dataset": "dataset-compatible-libraries"},
```
Comment (severo) [inline services/api/src/api/config.py:91]: why do we need this endpoint?

File: jobs/mongodb_migration/src/mongodb_migration/collector.py [hunk 2]
```diff
@@ -347,4 +347,14 @@ def get_migrations(self) -> list[Migration]:
...
+            QueueRenamingMigration(
+                version="20240307191001",
+                job_type="dataset-loading-tags",
+                new_job_type="dataset-compatible-libraries",
+            ),
+            CacheRenamingMigration(
+                cache_kind="dataset-loading-tags",
+                new_cache_kind="dataset-compatible-libraries",
+                version="20240307191000",
+            ),
```
Comment (severo) [inline jobs/mongodb_migration/src/mongodb_migration/collector.py:359]: maybe the metrics collections too

File: services/api/src/api/config.py (hunk 1 above)
Comment (lhoestq) [reply services/api/src/api/config.py:91]: For the front end to access the list of compatible libraries and the code snippets to show to the user.

Let me know if this endpoint is fine is if you have something else in mind

Comment (severo) [reply services/api/src/api/config.py:91]: Hmmm, OK, indeed, a lot of the data are not passed through sse-api (from `dataset-hub-cache`).

**Review by severo — APPROVED** [review]
see if you prefer to remove the metrics by hand, or through a migration, nit

**Comment by lhoestq** [comment]
by hand is fine :)

**Pull request merged**